tests/system/CommonFunctionsTest.php

```diff
@@ -137,6 +138,31 @@ public function testRedirectDefault()
         $this->assertInstanceOf(RedirectResponse::class, redirect());
     }
 
+    public function testRequestIncomingRequest()
+    {
+        Services::createRequest(new App());
+
+        $request = request();
+
+        $this->assertInstanceOf(IncomingRequest::class, $request);
```

Comment:
You should be able to assert that they reference the same object. 
```suggestion
        $this->assertInstanceOf(IncomingRequest::class, $request);
        $this->assertSame(Services::request(), $request);
```